Userland/Libraries/LibJS/Bytecode/Pass/GenerateCFG.cpp

```diff
@@ -0,0 +1,220 @@
+        auto const& instruction = *it;
+
+        if (instruction.type() == Instruction::Type::LeaveUnwindContext) {
+            if (unwind_frames.last().finalizer && unwind_frames.last().finalizer != &current_block)
+                dbgln("FIXME: Popping finalizer from the unwind context from outside the finalizer");
+            unwind_frames.take_last();
+
+            if (auto const* block = next_handler_or_finalizer())
+                enter_label(Label { *block }, current_block);
+        }
+
+        if (!instruction.is_terminator())
+            continue;
+
+        using enum Instruction::Type;
+        switch (instruction.type()) {
+        case Jump: {
+            auto true_target = *static_cast<Op::Jump const&>(instruction).true_target();
+            enter_label(true_target, current_block);
+            return;
+        }
+        case JumpIf:
+        case JumpNullish:
+        case JumpUndefined: {
+            // FIXME: It would be nice if we could avoid this copy, if we know that the unwind context stays the same in both paths
+            //        Or with a COW capable Vector alternative
+            // Note: We might partially unwind here, so we need to make a copy of
+            //       the current context to assure that the falsy code path has the same one
+            {
+                TemporaryChange saved_context { unwind_frames, unwind_frames };
+
+                auto true_target = *static_cast<Op::Jump const&>(instruction).true_target();
+                enter_label(true_target, current_block);
+            }
+
+            auto false_target = *static_cast<Op::Jump const&>(instruction).false_target();
+            enter_label(false_target, current_block);
+            return;
+        }
+        case Yield: {
+            auto continuation = static_cast<Op::Yield const&>(instruction).continuation();
+            if (continuation.has_value()) {
+                executable.exported_blocks->set(&continuation->block());
+                enter_label(*continuation, current_block);
+            } else if (auto const* finalizer = unwind_frames.last().finalizer) {
+                enter_label(Label { *finalizer }, current_block);
+                unwind_frames.last().finalizer_targets.append(nullptr);
+            }
+            return;
+        }
+        case Await: {
+            auto const& continuation = static_cast<Op::Await const&>(instruction).continuation();
+            executable.exported_blocks->set(&continuation.block());
+            enter_label(continuation, current_block);
+            return;
+        }
+        case EnterUnwindContext: {
+            auto entry_point = static_cast<Op::EnterUnwindContext const&>(instruction).entry_point();
+            auto handler_target = static_cast<Op::EnterUnwindContext const&>(instruction).handler();
+            auto finalizer_target = static_cast<Op::EnterUnwindContext const&>(instruction).finalizer();
```

Comment:
This information is not stored in the `EnterUnwindContext` instruction anymore, but is static in the block
Which should also simplify the CFG generation for those

Reply:
So uh, no clue how this compiles

Reply:
The bring-back-the-optimizers commit simply adds them back to `EnterUnwindContext` :sweat_smile: 
We can improve it separately :)